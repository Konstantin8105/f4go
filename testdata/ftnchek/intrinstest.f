      character *80 argmt
C  Test recognition of standard intrinsic
      x = sqrt(y)
C  Test mismatch of argument types
      z = sqrt(i)
      i = max(x,i)
C  Test recognition of extra intrinsics
      i = ior(i,8)
      i = not(i)
C  Test recognition of unix intrinsics
      call getarg(1,argmt)
C Test the different forms of rand and iargc: zero and one arg
      i = iargc()
      i = iargc(0)
      x = rand()
      x = rand(0)
C Test recognition of vms intrinsics
      x = ran(0)
      end

