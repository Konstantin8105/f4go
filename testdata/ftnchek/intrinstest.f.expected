IDENT               	character
*                   	*
INT                 	80
IDENT               	argmt
NEW_LINE            	

COMMENT             	C  Test recognition of standard intrinsic
NEW_LINE            	

IDENT               	x
=                   	=
IDENT               	sqrt
(                   	(
IDENT               	y
)                   	)
NEW_LINE            	

COMMENT             	C  Test mismatch of argument types
NEW_LINE            	

IDENT               	z
=                   	=
IDENT               	sqrt
(                   	(
IDENT               	i
)                   	)
NEW_LINE            	

IDENT               	i
=                   	=
IDENT               	max
(                   	(
IDENT               	x
,                   	,
IDENT               	i
)                   	)
NEW_LINE            	

COMMENT             	C  Test recognition of extra intrinsics
NEW_LINE            	

IDENT               	i
=                   	=
IDENT               	ior
(                   	(
IDENT               	i
,                   	,
INT                 	8
)                   	)
NEW_LINE            	

IDENT               	i
=                   	=
IDENT               	not
(                   	(
IDENT               	i
)                   	)
NEW_LINE            	

COMMENT             	C  Test recognition of unix intrinsics
NEW_LINE            	

CALL                	call
IDENT               	getarg
(                   	(
INT                 	1
,                   	,
IDENT               	argmt
)                   	)
NEW_LINE            	

COMMENT             	C Test the different forms of rand and iargc: zero and one arg
NEW_LINE            	

IDENT               	i
=                   	=
IDENT               	iargc
(                   	(
)                   	)
NEW_LINE            	

IDENT               	i
=                   	=
IDENT               	iargc
(                   	(
INT                 	0
)                   	)
NEW_LINE            	

IDENT               	x
=                   	=
IDENT               	rand
(                   	(
)                   	)
NEW_LINE            	

IDENT               	x
=                   	=
IDENT               	rand
(                   	(
INT                 	0
)                   	)
NEW_LINE            	

COMMENT             	C Test recognition of vms intrinsics
NEW_LINE            	

IDENT               	x
=                   	=
IDENT               	ran
(                   	(
INT                 	0
)                   	)
NEW_LINE            	

END                 	end
NEW_LINE            	

NEW_LINE            	

